hidparser/src/report_descriptor_parser.rs

```diff
@@ -146,9 +157,9 @@ impl ReportDescriptorParser {
         };
         feature_vec.push(ReportData::new(
           item.data,
-          &self.global_state[0],
-          &self.local_state,
-          &self.active_collections,
+          self.global_state[0].clone(),

```

Comment:
why did we switch from references to cloning? I was under the impression cloning is a last resort, which is why rust makes you specifically call `clone()`.

Reply:
I see you are just shifting the clone to the overall object rather than the individual pieces. That makes more sense now.

Reply:
Yes, I move the clone out of the `new` function so it become clearer when data are being clone. Also, if someone wants to create a struct by moving the content, now it is possible. In that case it would prevent some cloning.